### PR TITLE
RISC-V: upgrade CI image to use clang 17

### PR DESCRIPTION
**Pull request by mshabunin**
Updated image: https://github.com/opencv-infrastructure/opencv-gha-dockerfile/pull/38

### CODE REVIEW

**Comment by fengyuentau** [comment]
`OCV Nightly RISC-V / BuildAndTestNDS (4.x) (pull_request)` and `OCV Nightly RISC-V / BuildAndTestSC (4.x) (pull_request)` has the following initialization error:

```
Starting job container
  /usr/bin/docker pull quay.io/opencv-ci/opencv-ubuntu-22.04-riscv-nds:20240709
  Error response from daemon: unauthorized: access to the requested resource is not authorized
  Warning: Docker pull failed with exit code 1, back off 8.224 seconds before retry.
  /usr/bin/docker pull quay.io/opencv-ci/opencv-ubuntu-22.04-riscv-nds:20240709
  Error response from daemon: unauthorized: access to the requested resource is not authorized
  Warning: Docker pull failed with exit code 1, back off 6.869 seconds before retry.
  /usr/bin/docker pull quay.io/opencv-ci/opencv-ubuntu-22.04-riscv-nds:[20](https://github.com/opencv/ci-gha-workflow/actions/runs/9865139821/job/27241609060?pr=176#step:2:23)240709
  Error response from daemon: unauthorized: access to the requested resource is not authorized
  Error: Docker pull failed with exit code 1
```

---

`OCV Nightly RISC-V / BuildAndTestNDS (5.x) (pull_request)` and `OCV Nightly RISC-V / BuildAndTestSC (5.x) (pull_request)` are cancelled for some reasons.

File: .github/workflows/OCV-Nightly-RISCV.yaml [hunk 1]
```diff
@@ -345,3 +346,228 @@ jobs:
...
+      timeout-minutes: 60
+      run: |
+        cmake -G Ninja \
+          -S ${{ env.SRC_OPENCV }} \
+          -B ${{ env.BUILD_DIR }} \
+          -DCMAKE_TOOLCHAIN_FILE=$HOME/opencv/platforms/linux/riscv64-andes-gcc.toolchain.cmake \
+          -DRISCV_GCC_INSTALL_ROOT=/opt/andes \
+          -DCPU_BASELINE=RVV \
+          -DCPU_BASELINE_REQUIRE=RVV \
+          -DRISCV_RVV_SCALABLE=ON \
```
Comment (hanliutong) [inline .github/workflows/OCV-Nightly-RISCV.yaml:414]: Do we really want to enable and require RVV on the Andes platform? I'm not sure if the andes compiler/emulator supports RVV

Comment (mshabunin) [reply .github/workflows/OCV-Nightly-RISCV.yaml:414]: No, it's a mistake.

**Comment by mshabunin** [comment]
Builds are stable now. I had to disable some dnn tests in one of new configurations. Not sure what causes these issues.

**Review by fengyuentau — APPROVED** [review]

**Pull request merged**